src/components/PromotionList.tsx

```diff
@@ -43,7 +44,19 @@ export default function PromotionList({
         dispatch({ type: DispatchAction.DATA_FAILURE });
         setPromotions([]);
       });
-  }, [state.filter, state.sort, dispatch]);
+  }, [dispatch, state.filter, state.sort]);
+
+  /**
+   * When a search query is set, fetches promotions that satisfy the query.
+   */
+  useEffect(() => {
+    if (state.searchQuery) {
+      getPromotions([{ searchQuery: state.searchQuery }]).then((promotions) => {
+        dispatch({ type: DispatchAction.DATA_SUCCESS });
+        setPromotions(promotions);
+      });
+    }
+  }, [dispatch, state.searchQuery]);
```

Comment:
Pulled this into a separate useEffect, because after playing around with it a bit I realized that we'll probably want this code block to run only when the searchQuery changes. Previously, if we first used the search bar and then tried to use the dropdowns, it wasn't working because `state.SearchQuery` was defined (so it was always going into the if-then case).

Reply:
Ah good catch, that's a good point.